### PR TITLE
scripts: check epoch match for overrides files entry

### PR DESCRIPTION
**Pull request by dustymabe**
This should help us catch issues in the overrides file in CI. See
https://github.com/coreos/fedora-coreos-config/pull/293.

### CODE REVIEW

File: scripts/download-overrides.py [hunk 1]
```diff
@@ -20,34 +20,46 @@ def get_NVRA_from_NEVRA(string: str) -> str:
...
 for filename in os.listdir(os.path.join("src/config")):
     if is_override_lockfile(filename):
         with open(f'src/config/{filename}') as f:
             lockfile = yaml.safe_load(f)
         for pkg, pkgobj in lockfile['packages'].items():
-            rpms.add(get_NVRA_from_NEVRA(f"{pkg}-{pkgobj['evra']}"))
+            rpminfo = get_rpminfo(f"{pkg}-{pkgobj['evra']}")
+            rpmnvra = f"{rpminfo.name}-{rpminfo.version}-{rpminfo.release}.{rpminfo.arch}"
+            rpms.add(rpmnvra)
+            subprocess.check_call(['koji', 'download-build', '--rpm', rpmnvra], cwd='overrides/rpm')
```
Comment (jlebon) [inline scripts/download-overrides.py]: Hmm, if we used the Koji API directly we could verify the epoch before having to download the RPM. But meh... we're only continuously doing this in CI right now and not developers' laptops, so we can do that as a follow up!

File: scripts/download-overrides.py [hunk 2]
```diff
@@ -20,34 +20,46 @@ def get_NVRA_from_NEVRA(string: str) -> str:
...
+    return rpminfo
 
 def is_override_lockfile(filename: str) -> bool:
     return (filename.startswith('manifest-lock.overrides.')
             and filename[-4:] in ['json', 'yaml'])
 
+def assert_epochs_match(overrides_epoch: int, rpmfile_epoch: str):
+    # normalize the input into a string
+    if overrides_epoch is None:
+       normalized_overrides_epoch = '(none)'
```
Comment (jlebon) [inline scripts/download-overrides.py]: Worth mentioning here that this matches the output of `%{E}`.

File: scripts/download-overrides.py [hunk 3]
```diff
@@ -20,34 +20,46 @@ def get_NVRA_from_NEVRA(string: str) -> str:
...
+            rpmnvra = f"{rpminfo.name}-{rpminfo.version}-{rpminfo.release}.{rpminfo.arch}"
+            rpms.add(rpmnvra)
+            subprocess.check_call(['koji', 'download-build', '--rpm', rpmnvra], cwd='overrides/rpm')
+            # Make sure the epoch matches what was in the overrides file
+            # otherwise we can get errors: https://github.com/coreos/fedora-coreos-config/pull/293
+            cp = subprocess.run(['rpm', '-qp', '--queryformat', '%{E}', f'{rpmnvra}.rpm'],
+                                  check=True,
+                                  capture_output=True,
+                                  cwd='overrides/rpm')
+            rpmfile_epoch = cp.stdout.decode('utf-8')
```
Comment (jlebon) [inline scripts/download-overrides.py]: Optional: could fold into the previous run with e.g. `rpmfile_epoch = subprocess.check_output([...], ..., encoding='utf-8')`.

File: scripts/download-overrides.py (hunk 2 above)
Comment (dustymabe) [reply scripts/download-overrides.py]: +1

File: scripts/download-overrides.py (hunk 1 above)
Comment (dustymabe) [reply scripts/download-overrides.py]: +1 for follow up

File: scripts/download-overrides.py (hunk 3 above)
Comment (dustymabe) [reply scripts/download-overrides.py]: I've tested this like it is so I'll leave it for now

File: scripts/download-overrides.py (hunk 2 above)
Comment (dustymabe) [reply scripts/download-overrides.py]: ok I added a comment

**Pull request merged**